nixos/lib/test-driver/test_driver/machine.py

```diff
@@ -336,6 +338,10 @@ def __init__(
         self.name = name
         self.start_command = start_command
 
+        if out_dir is None:
+            out_dir = Path(".").resolve()
```

Comment:
I also thought about making the out-dir an optional parameter, defaulting to the PWD. Now the problem I thought of was, if a user runs the driver inside a directory without write-permission. The test-run will fail pretty late i.e. when `screenshot` or `copy_from_vm` is called. I found it nicer to fail as early as possible. The case can occur e.g. when someone does this:
```
nix-build . -A nixosTests.<test>.driver
cd result/bin
./nixos-test-driver
```
Here the result dir is a symlink to the nix store obviously...

Reply:
agree, but I think that check should be happening in the initialization of Driver, not in parsing the arguments. I rebased and pushed a new commit to fix this, please verify